walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -21,49 +21,102 @@
 public final class FindMeetingQuery {
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
     Collection<TimeRange> options = new ArrayList<>();
+    Collection<TimeRange> optionsWithOptional = new ArrayList<>();
     if(request.getDuration() > TimeRange.WHOLE_DAY.duration()) {
         return options;
     }
     int currentAvailableStart = TimeRange.START_OF_DAY;
     int currentAvailableEnd = TimeRange.END_OF_DAY;
     TimeRange currentAvailableRange = TimeRange.fromStartEnd(currentAvailableStart, currentAvailableEnd, true);
+    int currentAvailableStartOptional = TimeRange.START_OF_DAY;
+    int currentAvailableEndOptional = TimeRange.END_OF_DAY;
+    TimeRange currentAvailableRangeOptional = TimeRange.fromStartEnd(currentAvailableStartOptional, currentAvailableEndOptional, true);
     for(Event event : events) {
-        // If the event does not contain the requested attendees, go to the next event
-        Collection<String> groupOne = event.getAttendees();
-        Collection<String> groupTwo = request.getAttendees();
-        boolean hasAnyMatches = groupOne.stream().anyMatch(attendee1 -> groupTwo.stream().anyMatch(attendee2 -> attendee1.equals(attendee2)));
-        if(!hasAnyMatches) continue;
+        Collection<String> eventAttendees = event.getAttendees();
+        Collection<String> requiredAttendees = request.getAttendees();
+        Collection<String> optionalAttendees = request.getOptionalAttendees();
+        boolean isMandatory = eventAttendees.stream().anyMatch(attendee1 -> requiredAttendees.stream().anyMatch(attendee2 -> attendee1.equals(attendee2)));
+        boolean notAttending = !isMandatory && !eventAttendees.stream().anyMatch(attendee1 -> optionalAttendees.stream().anyMatch(attendee2 -> attendee1.equals(attendee2)));
```

Comment:
I really like the lambda usage here!

That said, can you limit your line length to max 120 characters?  Internally, we use different maximums depending on the project (often actually 100, rather than 120, but you can choose what you'd like).

Reply:
Ok sounds good